python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_condition_tests/test_auto_materialize_policy_conversion.py

```diff
@@ -0,0 +1,27 @@
+import dagster._check as check
+from dagster import (
+    AutoMaterializePolicy,
+    Definitions,
+    SchedulingCondition,
+    asset,
+    deserialize_value,
+    serialize_value,
+)
+
+
+def test_round_trip_conversion() -> None:
+    policy = SchedulingCondition.eager().as_auto_materialize_policy()
+    serialized_policy = serialize_value(policy)
+    deserialized_policy = deserialize_value(serialized_policy, AutoMaterializePolicy)
+    assert policy == deserialized_policy
+    assert deserialized_policy.asset_condition == SchedulingCondition.eager()
+
+
+def test_defs() -> None:
+    @asset(auto_materialize_policy=SchedulingCondition.eager().as_auto_materialize_policy())
```

Comment:
I think given this if I were a user the first think I would write is a shorthand function like `as_amp_policy`.

`auto_materialize_policy=AutoMaterializePolicy.from_scheduling_condition(SchedulingCondition.eager()` also an option

`@asset(auto_materialize_policy=as_amp_policy(SchedulingCondition.eager())`

I think the `as_auto_materialize_policy` might be a bit annoying to detail with in cases where there are complicated boolean experessions. I think it will be easy to make the mistake where you forgot to wrap the entire thing in parents.

Reply:
@schrockn "as amp policy" = "as auto materialize policy policy" (ATM machine-type thing). I think `as_amp()` might be too "informal".

`AutoMaterializePolicy.from_scheduling_condition` is possible for the user to do in this diff, but the drawback is they now need two imports (AutoMaterializePolicy and SchedulingCondition).

Considering we'd ideally want this construction to be fairly short-lived, I think a bit of verbosity isn't the end of the world, but open to pushback. Maybe the best solution is just to remove the `as_auto_materialize_policy()` thing and force users to go through `from_scheduling_condition()`?